### DISCUSSION TITLE
Tangent Offshoots in Free Draw Mode

### DISCUSSION THREAD

**Discussion by chonigman** [fabricjs/fabric.js#7228] [Q&A] [answered]
Hi!

Having a strange issue that occurs sporadically. Not sure if this is a bug or a configuration issue. When drawing in Free Draw mode if I draw a fast curved line there are these straight tangent offshoots that appear before I release the mouse button and the path is added to the canvas. The added  path is correct but the tangents are distracting when drawing. I have attached a video below. 

I am using the PencilBrush and setting a decimate  value to automatically smooth out lines. I have tried removing both which seems to work but then sometimes it just starts happening again and I switch them back on and again it's fine. 

I am building a drawing app inside a React app (not sure if that may cause weird intermittent issues like this). 

Thanks in advance. 

https://user-images.githubusercontent.com/1154580/126186501-9c5824ab-d187-4c47-add1-94e48ad4ce82.mp4

**Comment by i-prikot** [comment]
Hi.

When working with reactive frameworks, it's easy to overlook the reactivity of canvas or object properties. Perhaps some property of an object or canvas has become reactive.

**Comment by asturur** [comment]
Do you think you can replicate that in a jsfiddle? how is your brush configured?

**Reply by chonigman** [reply]
I will attempt to recreate in jsFiddle. The problem is intermittent and I cannot tie to any specific thing. When I made the jsFiddle the issue was happening. Then when I saved it stopped. 

I am using a PencilBrush and the decimate parameter to smooth out lines, like so, 

`canvas.freeDrawingBrush = new fabric.PencilBrush(canvas);`
`canvas.freeDrawingBrush.decimate = 3.0`

**Reply by asturur** [reply]
What version are you using of fabricJS? latest?

**Reply by chonigman** [reply]
Yes. 4.5.1

**Reply by chonigman** [reply]
Ok I have made a jsFiddle and I believe I know what is causing the issue. Please see the fiddle here, https://jsfiddle.net/chonigman/w7s1245g/60/

So, this is related to another issue I was having trying to draw a stroke indicator circle at the cursor tip. Currently, when the mouse moves, I have a callback that updates the stroke indicator to the mouse position. To make sure it updates the position I call canvas.requestRenderAll() and this call seems to be what's causing the tangent lines to be drawn.  They don't show up once the path is added to the canvas but if you draw a fast curved line the tangents stick out. 

While writing this I figured out the issue! I believe my solution is to just set the stroke indicator's opacity to 0 when in drawing mode and the mouse/touch is down and only make the requestRenderAll call on mouse move if not currently drawing a line. This does seem to be working now. Hopefully this weird edge case helps someone else in the future.

**Comment by chonigman** [comment, marked as answer]
Just for clarity, I figured out the problem, calling `requestRenderAll()` while in the middle of drawing a line in drawing mode. I was doing this because I was trying to update the position of a stroke indicator that I have to help show the user the size of the stroke they'll be drawing with. The solution was to only update the position of this stroke indicator when not actively drawing a line (but still in drawingMode).
